bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -145,6 +151,8 @@
     private final Set<String> loadedXmlThingTypes = new CopyOnWriteArraySet<>();
     private SafeCaller safeCaller;
     private volatile boolean active = false;
+    private StorageService storageService;
+    private volatile Storage<ManagedThingStatusInfo> storage;
```

Comment:
This can be a `Storage<Boolean>` see [here](https://github.com/kaikreuzer/smarthome/blob/circular_dependencies/bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngineImpl.java#L147) - and not need to declare it as volatile.